media-library-client/src/app/private-container/pages/home/video-gallery/video-tiles/video-tiles.tsx

```diff
@@ -39,33 +40,150 @@ export default function VideoTiles() {
+				<Grid item xs={6}>
+				<IconButtonWrapper id={'edit'} icon={IconSelector.Edit} sx={{ margin: 1, display: isEdit ? 'none' : 'inline-flex' }} onClick={() => setIsEdit(true)} />
+				<IconButtonWrapper id={'save'} icon={IconSelector.Save} sx={{ margin: 1, display: isEdit ? 'inline-flex' : 'none' }} onClick={() => {
+					if (videoGallery) {
+						HttpHelper.videoGallery.Save(videoGallery).then(res => alert(JSON.stringify(res))).catch(err => alert(JSON.stringify(err)));
+					}
+					setIsEdit(false)}} />
+				<IconButtonWrapper id={'cancel'} icon={IconSelector.Cancel} sx={{ margin: 1, display: isEdit ? 'inline-flex' : 'none' }} onClick={() => setIsEdit(false)} />
+
+					<ValidatedTextField
+						id="name"
+						label=""
+						value={`${videoGallery?.name}`}
+						isReadOnly={!isEdit}
+						onChange={value => {
+							setVideoGallery(prevState => {
+								if (!prevState) return prevState;
+								const temp = { ...prevState };
+								temp.name = value;
+								return temp;
+							});
+						}}
+					/>
+
+					<Rating
+						name="no-value"
+						sx={{ mb: 1, paddingTop: 1, marginLeft: 1 }}
+						value={videoGallery?.rating ?? 0}
+						onChange={(event, newValue) => {
+							if (videoGallery && newValue) {
+								videoGallery.rating = newValue;
+								videoGallery.status = ObjectStatus.Modified;
+								setVideoGallery(videoGallery);
+								HttpHelper.videoGallery
+									.Save(videoGallery)
+									.then(res => alert(`Rating has been updated`))
+									.catch(err => alert(`Save has been unsuccessful due to: ${err}`));
+								videoGallery.status = ObjectStatus.None;
+							}
+						}}
+					/>
+					<ValidatedTextField
+						id="series description"
+						label="Description"
+						value={videoGallery?.description}
+						isReadOnly={!isEdit}
+						onChange={value => {
+							setVideoGallery(prevState => {
+								if (!prevState) return prevState;
+								const temp = { ...prevState };
+								temp.description = value;
+								return temp;
+							});
+						}}
+					/>
+					<NavItem title="My Anime List" href={`${videoGallery?.wikiLink}`} />
+					<ValidatedTextField
+						id="airing state"
+						label="Airing State"
+						value={`${videoGallery?.airingState}`}
+						isReadOnly={!isEdit}
+						onChange={value => {
+							setVideoGallery(prevState => {
+								if (!prevState) return prevState;
+								const temp = { ...prevState };
+								temp.airingState = +value;
+								return temp;
+							});
+						}}
+					/>
+					<ValidatedTextField
+						id="episode count"
+						label="Episode Count"
+						value={`${videoGallery?.episodeCount}`}
+						isReadOnly={!isEdit}
+						onChange={value => {
+							setVideoGallery(prevState => {
+								if (!prevState) return prevState;
+								const temp = { ...prevState };
+								temp.episodeCount = +value;
+								return temp;
+							});
+						}}
+					/>
+					<Typography
+						onClick={() => {

```

Comment:
Is this click event good enough? Should it rather be on the entire grid item filterd on/off based on the edit state

Reply:
We should probably discuss this. Since this should technically be a list of videos